File: app/src/main/java/ru/samtakoy/features/import_export/QPacksExporter.java
package ru.samtakoy.features.import_export;

import io.reactivex.Completable;
import ru.samtakoy.core.data.local.database.room.entities.QPackEntity;

public interface QPacksExporter {

    //void exportThemeTree(QPack qPack);
    Completable exportQPackToEmailRx(QPackEntity qPack);

    Completable exportQPack(QPackEntity qPack);


    Completable exportAllToEmail();

    Completable exportAllToFolder(String exportDirPath);

}

File: app/src/main/java/ru/samtakoy/features/import_export/QPacksExporter.kt
package ru.samtakoy.features.import_export

import androidx.annotation.WorkerThread
import ru.samtakoy.core.data.local.database.room.entities.QPackEntity

interface QPacksExporter {
    @WorkerThread
    suspend fun exportQPackToEmail(qPackId: Long)

    @WorkerThread
    suspend fun exportQPack(qPack: QPackEntity)

    @WorkerThread
    suspend fun exportAllToEmail()

    @WorkerThread
    suspend fun exportAllToFolder(exportDirPath: String)
}
